script.js

```diff
@@ -1,76 +1,228 @@
+}
+repeatStringNumTimes("*", 8);
+repeatStringNumTimes("abc", 3)
+
+// task 8 Truncate a String
+function truncateString(str, num) {
+  if (str.length > num) {
+    return console.log(`Task 8 result : ${str.slice(0, num)}...`);
+  } else {
+    return console.log(`Task 8 result : ${str.slice(0, num)}`);
+  }
+}
+truncateString("A-tisket a-tasket A green and yellow basket", 8);
+
+// task 9 Finders Keepers
+function findElement(arr, func) {
+  let num = 0;
+  for (let i = 0; i < arr.length; i++) {
+    if (func(arr[i])) {
+      num = arr[i];
+      break;
+    } else {
+      num = undefined;
+    }
+  }
+  return console.log(`Task 9 result : ${num}`);
+}
+findElement([1, 3, 5, 8, 9, 10], function (num) {
+  return num % 2 === 0;
+});
+findElement([1, 3, 5, 9], function (num) {
+  return num % 2 === 0;
+});
+
+// task 10 Boo who
+function booWho(bool) {
+    if (typeof bool ==='boolean') bool = true
+    else bool = false
+    return console.log(`Task 10 result : ${bool}`)  
+  }
+  booWho(true)
+  booWho([1, 2, 3])
+
+  // task 11 Title Case a Sentence
+  function titleCase(str) {
+    str = str.toLowerCase()
+    str = str.split(' ').map((item) => item[0].toUpperCase()+item.slice(1)).join(' ')
+    return console.log(`Task 11 result : ${str}`)
+  }
+  titleCase("I'm a little tea pot")
+  titleCase("sHoRt AnD sToUt")
+
+  // task 12 Slice and Splice
+  function frankenSplice(arr1, arr2, n) {
+    const clonArr2 = arr2.slice(0)
+    for(let i = 0 ; i<arr1.length; i++){
+      clonArr2.splice(n , 0 , arr1[i])
+      n++
+    }
+    return console.log(`Task 12 result : ${clonArr2}`);
+  }
+  frankenSplice([1, 2, 3], [4, 5, 6], 1);
+  frankenSplice([1, 2], ["a", "b"], 1)
+  frankenSplice(["claw", "tentacle"], ["head", "shoulders", "knees", "toes"], 2)
+  frankenSplice([1, 2, 3, 4], [], 0)
+
+  // task 13 Falsy Bouncer
+  function bouncer(arr) {
+    arr = arr.filter((item) =>{
+      if(item !== "" || item !== false ||
+       item !== null || item !== 0 ||
+       item !== undefined|| item !== NaN){
+        return item
+      }
     })
-    return console.log(largeNumberArr)
+    return console.log(`Task 13 result : ${arr}`)
+    
   }
-  
-  largestOfFour([[17, 23, 25, 12], [25, 7, 34, 48], [4, -10, 18, 21], [-72, -3, -17, -10]])
+  bouncer([7, "ate", "", false, 9]);
+  bouncer(["a", "b", "c"])
+  bouncer([false, null, 0, NaN, undefined, ""])
+  bouncer([null, NaN, 1, 2, undefined])
+
+  // task 14 Where do I Belong
+  function getIndexToIns(arr, num) {
+    arr.sort((a , b)=> a-b)
+   for(let i = 0 ; i < arr.length ; i++){
+    if(arr[i] >= num ){
+      num = i
+      break
+    }
 
+   }
+    return console.log(`Task 14 result : ${num}`);
+  }
+  getIndexToIns([10, 20, 30, 40, 50], 35)
+  getIndexToIns([10, 20, 30, 40, 50], 30)
+  getIndexToIns([5, 3, 20, 3], 5)
+  getIndexToIns([40, 60], 50)
+  getIndexToIns([3, 10, 5], 3)
 
+  // task 15 Mutations
+  function mutation(arr) {
+    let result
```

Comment:
function mutation(arr) {
    const firstStr = arr[0].toLowerCase()
    const seconStr = arr[1].toLowerCase()
    for(let i = 0; i < seconStr.length ; i++){
      if(!firstStr.includes(seconStr[i])) {
        return false
      }
  }
  return true
}